proxy/connect/connect.go

```diff
@@ -131,26 +136,43 @@ sendLoop:
 	for {
 		select {
 		case request := <-d.sendChannel:
+			select {
+			case <-d.statsTicker.C:
+				destinationTag := fmt.Sprintf("destination:%s", d.address)
+				d.statsd.Gauge(
+					"veneur_proxy.forward.buffer_size",
+					float64(len(d.sendChannel)), []string{destinationTag}, 1.0)
+				d.statsd.Gauge(
+					"veneur_proxy.forward.buffer_latency_ns",
+					float64(time.Since(request.Timestamp).Nanoseconds()),
+					[]string{destinationTag}, 1.0)
+			default:
+				// Do nothing.
+			}
 			err := d.client.Send(request.Metric)
-			if err != nil {
+			if err == io.EOF {
 				d.logger.WithError(err).Debug("failed to forward metric")
 				d.statsd.Count(
 					"veneur_proxy.forward.metrics_count", 1,
-					[]string{"error:true"}, 1.0)
+					[]string{"error:eof"}, 1.0)
+				break sendLoop
+			} else if err != nil {
+				d.logger.WithError(err).Debug("failed to forward metric")
+				d.statsd.Count(
+					"veneur_proxy.forward.metrics_count", 1,
+					[]string{"error:forward"}, 1.0)
 			} else {
 				d.statsd.Count(
 					"veneur_proxy.forward.metrics_count", 1,
 					[]string{"error:false"}, 1.0)
 			}
-			request.ErrorChannel <- err
-			close(request.ErrorChannel)
 		case <-ctx.Done():
 			break sendLoop
 		}
 	}
 
-	close(d.sendChannel)
 	d.destinationHash.RemoveDestination(d.address)
+	close(d.sendChannel)
```

Comment:
It is usually discouraged to close a channel from the receiver side. This could lead to panic in the send loop